src/Preset.php

```diff
@@ -146,4 +168,35 @@ protected static function updatePackageScripts()
         );
     }
 
+    protected static function updateComposer($dev = true)
+    {
+        if (! file_exists(base_path('composer.json'))) {
+            return;
+        }
+
+        $configurationKey = $dev ? 'require' : 'require-dev';
+
+        $composer = json_decode(file_get_contents(base_path('composer.json')), true);
+
+        $composer[$configurationKey] = static::updateComposerArray(
+            array_key_exists($configurationKey, $composer) ? $composer[$configurationKey] : [],
+            $configurationKey
+        );
+
+        ksort($composer[$configurationKey]);
+
+        file_put_contents(
+            base_path('composer.json'),
+            json_encode($composer, JSON_UNESCAPED_SLASHES | JSON_PRETTY_PRINT).PHP_EOL
+        );
+    }
+
+    // update the composer JSON array
+    protected static function updateComposerArray(array $packages)
+    {
+        return array_merge(
+            static::COMPOSER_DEV_PACKAGES_TO_ADD,
+            Arr::except($packages, static::COMPOSER_DEV_PACKAGES_TO_ADD)
```

Comment:
We can reverse the order of the function parameters to prevent $packages overwrites our Dev packages. 
The `Arr::except` in the original `updatePackages()` is mainly for case that we have some packages want to be removed from the original dependency.

Reply:
👍